projects/map-ol/src/lib/map-ol.service.ts

```diff
@@ -73,13 +73,28 @@ const TITLE_KEY = 'title';
 const WebMercator = 'EPSG:3857';
 const WGS84 = 'EPSG:4326';
 
+/**
+ * While @dlr-eoc/services-layers.popup already contains instructions about how to build a popup,
+ * IPopupArgs adds additional, map-ol-specific context like the olLayer, the feature, and the olMapBrowserEvent.
+ */
 export interface IPopupArgs {
   modelName: string;
   properties: any;
   layer: olLayer<any>;
   feature?: olFeature<any> | olRenderFeature;
   event: olMapBrowserEvent<PointerEvent>;
   popupFn?: popup['pupupFunktion'];
+  dynamicPopup?: popup['dynamicPopup'];
+}
+
+/**
+ * In dynamic popups we want even more context-information than in static popups.
+ * Most notably, we want @dlr-eoc/services-layers.popup.filterkeys and
+ * @dlr-eoc/services-layers.popup.properties to be available, too.
+ */
+export interface IDynamicPopupArgs extends IPopupArgs {
```

Comment:
normally the `popupFn` is not needed here, same for `modelName` and `event`.

The interfaces `IPopupArgs` and [`popup`](https://github.com/dlr-eoc/ukis-frontend-libraries/blob/v7.2.0/projects/services-layers/src/lib/types/Layers.ts#L31) were already not quite right from the beginning.
It should just be possible to use either `plain properties` or `popupFn` or `asyncPupup` or `dynamicPopup`.
These can be combined with the filters/options:
- `filterkeys`  /** limit properties */ 
- `properties` /** To overwrite the keys of the layer properties e.g.   titleDe  to Title De*/ 
- `single` /** default is false - removes the other popups if the next is added */
- `options` /** options which get assigned on the popup creations e.g. ol/Overlay autoPan: false */

but these are already handled before creation of the *popup body*.

We can try to change this if we have further breaking changes!

Reply:
Alright, thanks a lot for the explanation!
I've added some additional comments by @boeckMt 's  as [comments](https://github.com/dlr-eoc/ukis-frontend-libraries/pull/63/commits/3299044d3d831ccddf046c31a9170fce3c7d02d7) to the code.